### ISSUE TITLE
Questions about amp

### ISSUE THREAD

**Issue by zs963048949** [pytorch/pytorch#50941]
I want to use torch.cuda.amp.autocast to reduce the GPU memory，but I got a odd phenomenon.
The GPU memory will be decreased，only when I add the following code in forward() function of nn.Module class:
```
    # @autocast()
    def forward(self, x):
        with autocast():
            x1 = self.conv(x)
            out = self.outc(x1)
        return out  
```
- I delete other code about amp.autocast and GradScaler in the train.py,and then the GPU memory still can be decreased from 9000+ to 6000.
- In addition,I can not control the GPU number,when I use "@autocast()". I have set one GPU,but all GPU were used.
Thanks for your help!
pytorch version:1.6.0+cu101
python version:3.65

**Comment by zs963048949** [comment]
### Supplement:
It will happens,when I use *WithLogitsLoss() as loss function.If not,the error message will be found:
**Found dtype Float but expected Half**
